engine/src/test/java/org/camunda/bpm/engine/test/api/runtime/ModificationExecutionSyncTest.java

```diff
@@ -97,6 +102,61 @@ public void createSimpleModificationPlan() {
+    List<String> instances = helper.startInstances("process1", 2);
+    HistoricProcessInstanceQuery historicProcessInstanceQuery = historyService.createHistoricProcessInstanceQuery();
+    historicProcessInstanceQuery.processDefinitionId(processDefinition.getId());
+
+    runtimeService.createModification(processDefinition.getId()).startBeforeActivity("user2")
+        .cancelAllForActivity("user1").historicProcessInstanceQuery(historicProcessInstanceQuery).execute();
+
+    for (String instanceId : instances) {
+      List<String> activeActivityIds = runtimeService.getActiveActivityIds(instanceId);
+      assertEquals(1, activeActivityIds.size());
+      assertEquals(activeActivityIds.iterator().next(), "user2");
+    }
+  }
+
+  @Test
+  public void createSimpleModificationPlanWithIdenticalRuntimeAndHistoryQuery() {
+    ProcessDefinition processDefinition = testRule.deployAndGetDefinition(instance);
+    List<String> instances = helper.startInstances("process1", 2);
+    HistoricProcessInstanceQuery historicProcessInstanceQuery = historyService.createHistoricProcessInstanceQuery();
+    historicProcessInstanceQuery.processDefinitionId(processDefinition.getId());
+    ProcessInstanceQuery processInstanceQuery = runtimeService.createProcessInstanceQuery();
+    processInstanceQuery.processDefinitionId(processDefinition.getId());
+
+    runtimeService.createModification(processDefinition.getId()).startBeforeActivity("user2")
+        .cancelAllForActivity("user1").historicProcessInstanceQuery(historicProcessInstanceQuery).processInstanceQuery(processInstanceQuery).execute();
+
+    for (String instanceId : instances) {
+      List<String> activeActivityIds = runtimeService.getActiveActivityIds(instanceId);
+      assertEquals(1, activeActivityIds.size());
+      assertEquals(activeActivityIds.iterator().next(), "user2");
+    }
+  }
+
+  @Test
+  public void createSimpleModificationPlanWithComplementaryRuntimeAndHistoryQueries() {
+    ProcessDefinition processDefinition = testRule.deployAndGetDefinition(instance);
+    List<String> instances = helper.startInstances("process1", 2);
+    HistoricProcessInstanceQuery historicProcessInstanceQuery = historyService.createHistoricProcessInstanceQuery();
+    historicProcessInstanceQuery.processInstanceId(instances.get(0));
+    ProcessInstanceQuery processInstanceQuery = runtimeService.createProcessInstanceQuery();
+    processInstanceQuery.processInstanceId(instances.get(1));
+
+    runtimeService.createModification(processDefinition.getId()).startBeforeActivity("user2")
+        .cancelAllForActivity("user1").historicProcessInstanceQuery(historicProcessInstanceQuery).processInstanceQuery(processInstanceQuery).execute();
+
+    for (String instanceId : instances) {
+      List<String> activeActivityIds = runtimeService.getActiveActivityIds(instanceId);
+      assertEquals(1, activeActivityIds.size());
+      assertEquals(activeActivityIds.iterator().next(), "user2");
+    }
+  }
+
```

Comment:
✍️ One more test case with filter unfinished:
```suggestion

  @Test
  public void createSimpleModificationPlanWithHistoricQueryUnfinished() {
    ProcessDefinition processDefinition = testRule.deployAndGetDefinition(instance);
    List<String> instances = helper.startInstances("process1", 2);
    HistoricProcessInstanceQuery historicProcessInstanceQuery = historyService.createHistoricProcessInstanceQuery();
    historicProcessInstanceQuery.processDefinitionId(processDefinition.getId()).unfinished();

    runtimeService.deleteProcessInstance(instances.get(0), "test");

    runtimeService.createModification(processDefinition.getId()).startBeforeActivity("user2")
        .cancelAllForActivity("user1").historicProcessInstanceQuery(historicProcessInstanceQuery).execute();

    List<String> activeActivityIds = runtimeService.getActiveActivityIds(instances.get(1));
    assertEquals(1, activeActivityIds.size());
    assertEquals(activeActivityIds.iterator().next(), "user2");
  }

```